src/zed_wrapper_node.cpp

```diff
@@ -203,143 +277,199 @@ int main(int argc, char **argv) {
+                        }
                     }
                     continue;
                 }
 
                 old_t = ros::Time::now();
 
-                // Retrieve RGBA Left image
-                slMat2cvMat(zed->retrieveImage(sl::zed::SIDE::LEFT)).copyTo(leftImRGBA);
-                // Convert to RGB
-                cv::cvtColor(leftImRGBA, leftImRGB, CV_RGBA2RGB);
+                // Publish the left == rgb image if someone has subscribed to
+                if(left_SubNumber>0 || rgb_SubNumber>0) {
+                    // Retrieve RGBA Left image
+                    cv::cvtColor(slMat2cvMat(zed->retrieveImage(sl::zed::SIDE::LEFT)), leftImRGB, CV_RGBA2RGB); // Convert to RGB
+                    if(left_SubNumber>0) {
+                        publishCamInfo(left_cam_info_msg, pub_left_cam_info, t);
+                        publishImage(leftImRGB, pub_left, left_frame_id, t);
+                    }
+                    if(rgb_SubNumber>0) {
+                        publishCamInfo(rgb_cam_info_msg, pub_rgb_cam_info, t);
+                        publishImage(leftImRGB, pub_rgb, rgb_frame_id, t); // rgb is the left image
+                    }
+                }
+                
+                // Publish the right image if someone has subscribed to
+                if(right_SubNumber>0) {
+                    // Retrieve RGBA Right image
+                    cv::cvtColor(slMat2cvMat(zed->retrieveImage(sl::zed::SIDE::RIGHT)), rightImRGB, CV_RGBA2RGB); // Convert to RGB
+                    publishCamInfo(right_cam_info_msg, pub_right_cam_info, t);
+                    publishImage(rightImRGB, pub_right, right_frame_id, t);
+                }
 
-                if (computeDepth) {
-                    ROS_DEBUG_STREAM("Publishing left and depth Images");
+                // Publish the depth image if someone has subscribed to
+                if (depth_SubNumber>0) {
                     // Retrieve raw depth data and convert it to 16_bit data
-                    slMat2cvMat(zed->retrieveMeasure(sl::zed::MEASURE::DEPTH)).convertTo(secondIm, CV_16UC1);
-                } else {
-                    ROS_DEBUG_STREAM("Publishing left and right Images");
-                    // Retrieve RGBA Right image
-                    slMat2cvMat(zed->retrieveImage(sl::zed::SIDE::RIGHT)).copyTo(secondImRaw);
-                    // Convert to RGB
-                    cv::cvtColor(secondImRaw, secondIm, CV_RGBA2RGB);
+                    slMat2cvMat(zed->retrieveMeasure(sl::zed::MEASURE::DEPTH)).convertTo(depthIm, CV_16UC1);
+                    publishDepth(depthIm, pub_depth, depth_frame_id, t);
+                }
+                
+                // Publish the point cloud if someone has subscribed to
+                if (cloud_SubNumber>0 && pointCloudThreadRunning == false) {
+                    // Retrieve raw pointCloud data
+                    float* cloud = (float*) zed->retrieveMeasure(sl::zed::MEASURE::XYZRGBA).data;
```

Comment:
From the documentation:

> The buffer must be duplicated if another CPU retrieve has to be done, it will be overwritten otherwise.
> The buffer must not be freed.

Is it possible that the main thread is wiping this buffer before the background thread is done with it?